### PR TITLE
Allow getting assets by BAR build ID via `darc get-asset`

### PR DESCRIPTION
**Pull request by premun**
https://github.com/dotnet/arcade-services/issues/2600

### Release Note Category
- [x] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements

### Release Note Description
Allow getting all assets belonging to a build via `darc get-asset`

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/Darc/Operations/GetAssetOperation.cs [hunk 1]
```diff
@@ -49,7 +55,12 @@ public override async Task<int> ExecuteAsync()
             List<Asset> matchingAssets =
                 (await remote.GetAssetsAsync(name: _options.Name, version: _options.Version, buildId: _options.Build)).ToList();
 
-            var queryDescription = new StringBuilder($"name '{_options.Name}'");
+            var queryDescription = new StringBuilder();
+
+            if (!string.IsNullOrEmpty(_options.Name))
+            {
+                queryDescription.Append($" named '{_options.Name}'");
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/Darc/Operations/GetAssetOperation.cs:62]: If the command is run without this option, and with the build, the message will be a bit weird, for example `Looking up assets and version 1.0.0...`

Comment (dkurepa) [reply src/Microsoft.DotNet.Darc/Darc/Operations/GetAssetOperation.cs:62]: Maybe just rearranging the `ifs` a bit, swapping the positions of the build and versions one

Comment (premun) [reply src/Microsoft.DotNet.Darc/Darc/Operations/GetAssetOperation.cs:62]: Fixed

**Review by dkurepa — DISMISSED** [review]

File: src/Microsoft.DotNet.Darc/Darc/Operations/GetAssetOperation.cs [hunk 2]
```diff
@@ -49,29 +55,37 @@ public override async Task<int> ExecuteAsync()
...
+                queryDescription.Append($" with version '{_options.Version}'");
             }
 
             if (targetChannel != null)
             {
                 queryDescription.Append($" on channel '{targetChannel.Name}'");
             }
 
-            if (_options.Build != null)
+            if (!_options.Build.HasValue)
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Darc/Darc/Operations/GetAssetOperation.cs:80]: This should be options.maxAge

Comment (premun) [reply src/Microsoft.DotNet.Darc/Darc/Operations/GetAssetOperation.cs:80]: I don't think so - we want to remove the age filter when we pass in the build

Comment (dkurepa) [reply src/Microsoft.DotNet.Darc/Darc/Operations/GetAssetOperation.cs:80]: Ahh right, and the MaxAgeInDays has a default has a default value, so we don't have to check it

**Review by dkurepa — APPROVED** [review]

**Pull request merged**